executor/distsql.go

```diff
@@ -386,6 +389,38 @@ func (e *IndexLookUpExecutor) Open(ctx context.Context) error {
 	return err
 }
 
+func (e *IndexLookUpExecutor) buildTableKeyRanges() (err error) {
+	sc := e.ctx.GetSessionVars().StmtCtx
+	if e.partitionTableMode {
+		if e.keepOrder { // this case should be prevented by the optimizer
+			return errors.New("invalid execution plan: cannot keep order when accessing a partition table by IndexLookUpReader")
+		}
+		e.feedback.Invalidate() // feedback for partition tables is not ready
+		e.partitionKVRanges = make([][]kv.KeyRange, 0, len(e.prunedPartitions))
+		for _, p := range e.prunedPartitions {
+			physicalID := p.GetPhysicalID()
+			var kvRange []kv.KeyRange
+			if e.index.ID == -1 {
+				kvRange, err = distsql.CommonHandleRangesToKVRanges(sc, []int64{physicalID}, e.ranges)
+			} else {
+				kvRange, err = distsql.IndexRangesToKVRanges(sc, physicalID, e.index.ID, e.ranges, e.feedback)
```

Comment:
The API design is so weird. One of them support multiple partitions and the other does not.

Reply:
Yes, actually there is a function `IndexRangesToKVRangesForTables` that can handle multiple partitions at a time, but we cannot use it since it would mix all `kvRanges` from different partitions up so that we cannot distinguish them then.
`CommonHandleRangesToKVRanges` has the same problem, so I only have to build `kvRange` one by one.
Could you give me any suggestion to solve this?